src/main/kotlin/org/opensearch/commons/alerting/action/DocLevelMonitorFanOutRequest.kt

```diff
@@ -29,6 +39,104 @@ class DocLevelMonitorFanOutRequest : ActionRequest, ToXContentObject {
     val concreteIndicesSeenSoFar: List<String>
     val workflowRunContext: WorkflowRunContext?
 
+    companion object {
+        val log = LogManager.getLogger(DocLevelMonitorFanOutRequest::class.java)
+        private fun safeReadMonitor(sin: StreamInput): Monitor =
+            try {
+                Monitor.readFrom(sin)!!
+            } catch (e: Exception) {
+                log.error("Error parsing monitor in Doc level monitor fanout request", e)
+                Monitor(
+                    "failed_serde", NO_VERSION, "failed_serde", true,
+                    IntervalSchedule(1, ChronoUnit.MINUTES), Instant.now(), Instant.now(), "",
+                    null, NO_SCHEMA_VERSION, emptyList(), emptyList(), emptyMap(),
+                    DataSources(), false, false, "failed"
+                )
+            }
+
+        private fun safeReadBoolean(sin: StreamInput): Boolean =
+            try {
+                sin.readBoolean()
+            } catch (e: Exception) {
+                log.error("Error parsing boolean in Doc level monitor fanout request", e)
+                false
```

Comment:
How does it affect execution if we end up with the default value for these fields? It seems like we need some way to signal that the fanout should not execute if we encounter a deserialization error like this

Reply:
we are not gonna use these fields. the alerting plugin will look for null check on index execution context https://github.com/opensearch-project/alerting/pull/1815

Reply:
added a flag to indicate serialization failed if any of the safeRead*() methods fail